Mathlib/Topology/Algebra/ContinuousMonoidHom.lean

```diff
@@ -404,3 +406,368 @@ theorem locallyCompactSpace_of_hasBasis (V : ℕ → Set Y)
 end LocallyCompact
 
 end ContinuousMonoidHom
+
+end
+
+section
+
+/-!
+
+# Continuous MulEquiv
+
+This section defines the space of continuous isomorphisms between two topological groups.
+
+## Main definitions
+
+-/
+
+universe u v
+
+variable (G : Type u) [TopologicalSpace G] (H : Type v) [TopologicalSpace H]
+
+/-- Define the structure of two-sided continuous isomorphism of additive groups. -/
+structure ContinuousAddEquiv [Add G] [Add H] extends AddEquiv G H , Homeomorph G H
+
+/-- Define the structure of two-sided continuous isomorphism of groups. -/
+@[to_additive "The type of two-sided continuous isomorphism of additive groups."]
+structure ContinuousMulEquiv [Mul G] [Mul H] extends MulEquiv G H , Homeomorph G H
+
+/-- The Homeomorphism induced from a two-sided continuous isomorphism of groups. -/
+add_decl_doc ContinuousMulEquiv.toHomeomorph
+
+/-- The Homeomorphism induced from a two-sided continuous isomorphism additive groups. -/
+add_decl_doc ContinuousAddEquiv.toHomeomorph
+
+/-- Notation for a `ContinuousMulEquiv`. -/
+infixl:25 " ≃ₜ* " => ContinuousMulEquiv
+
+/-- Notation for an `ContinuousAddEquiv`. -/
+infixl:25 " ≃ₜ+ " => ContinuousAddEquiv
+
+section
+
+/-- `ContinuousAddEquivClass F A B` states that `F` is a type of two-sided continuous additive
+isomomorphisms.-/
+class ContinuousAddEquivClass (F : Type*) (A B : outParam Type*) [Add A] [Add B]
+    [h : EquivLike F A B] [TopologicalSpace A] [TopologicalSpace B] extends AddEquivClass F A B :
```

Comment:
Please avoid hom classes that use both algebra and topology. A better way is to introduce `HomeomorphClass` and assume `[AddEquivClass F A B] [HomeomorphClass F A B]` whenever needed.

Reply:
I think `HomeomorphClass` doesn't exist at all. 
Also, the design is under the suggextion below:
![图片](https://github.com/user-attachments/assets/ba5a13e0-0d18-4493-ad08-426d94c189da)
Is there a reason of avoiding the use of both algebra and topology?
Sorry that I donesn't know much about how to design APIs, I just follow the suggestions copying the existing codes and create this PR.

Reply:
Also, what about `ContinuousMonoidHomClass`, in this sense why this exists?

Reply:
@Thmoas-Guan earlier in Mathlib, we would mix algebra / topology in hom classes. This was necessary because they all extended `FunLike` (or `EquivLike`). However, we have since pulled this out and pass it as an argument instead. As such, the hom classes are now `Prop`-valued, and it now makes more sense to split them into their respective parts, hence the suggestion from Yury not to mix.

Also, Yury knows we don't have `HomeomorphClass`, that's why he's asking you to introduce it (it should be a separate PR though, on which this one will then depend). Note also: `ContinuousMonoidHomClass` is deprecated.

Does this course of action make sense? Do you know how to proceed?

Reply:
Sorry that I didn't got up to date with the latest mathlib and didn't notice it.
The part I really needed is only `symm` and `trans`, so maybe I can delete the `class` instances in this file for now and start a new PR (I think independent of this) to work on the class of `Homeomorph` and add class instances to this file?

Reply:
I introduced `HomeomorphClass` in PR #18689, may I ask for your help to review it too? Thank you